liferay-echo/liferay-portal
# LPS-170032 Page with background Image is not rendered in remote site

## Motivation

[Link to task](https://issues.liferay.com/browse/LPS-170032)

Remote staging fails when a container has a link or background image associated to it. This is because we are not taking the mapped values in the structure when exporting/importing, only those in fragment entry links

## Proposed solution

@victorg1991 already defined a method for taking mapped values into account on export/import, so apply his solution for layout structures too


## Change summary:

* Move existing code for collection to content processors
* Also add logic for container mapped values (background image and links) to content processors
* Adds integration test


## Test Scenario (full)

* Create a new site
* Updload three different documents
* Create a new widget page that will serve as link
* Create a new content page
* Add 4 containers to the page for each case scenario
* For the first container, map the link to the first document -> display page URL
* For the second container, map the link to the widget page
* For the third container, select as mapped background the second document -> previewImage
* For the fourth container, select as direct background image the third document
* Publish the page
* Go to Publishing -> Export, and export the whole site, download the LAR
* Stop the portal and delete the database
* Start the portal and create a new site
* Go to Publishing -> Import, and import the LAR
* The page should be available and have the 4 containers with the configured values
